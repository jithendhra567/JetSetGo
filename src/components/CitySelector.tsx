import React from 'react';
import {FlatList, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {COLORS, FONT_SIZES} from '../utils/styles';
import {WIDTH} from '../utils/common';

export const CITIES = [
  'Mumbai',
  'Delhi',
  'Bangalore',
  'Chennai',
  'Kolkata',
  'Hyderabad',
  'Vishakapatanam',
];

export type selectType = 'from' | 'to';

type Props = {
  selectionKey: selectType;
  onClose: () => void;
  onSelect: (value: string) => void;
};

const CitySelector = ({selectionKey, onClose, onSelect}: Props) => {
  const onItemPress = (item: string) => {
    onSelect(item);
    onClose();
  };

  const renderItem = ({item}: {item: string}) => (
    <TouchableOpacity
      onPress={() => onItemPress(item)}
      style={styles.itemContainer}>
      <Text style={styles.item}>{item}</Text>
    </TouchableOpacity>
  );

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.title}>Travelling {selectionKey}?</Text>
        <TouchableOpacity onPress={onClose}>
          <Text style={styles.close}>Close</Text>
        </TouchableOpacity>
      </View>
      <FlatList
        data={CITIES}
        renderItem={renderItem}
        keyExtractor={item => item}
        // ItemSeparatorComponent={() => <View style={styles.seperator} />}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: WIDTH,
    padding: 20,
    backgroundColor: COLORS.WHITE,
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 20,
  },
  title: {
    color: COLORS.BLACK,
    fontSize: FONT_SIZES['3xl'],
    fontWeight: 'bold',
    textTransform: 'capitalize',
  },
  close: {
    color: COLORS.RED,
    fontSize: FONT_SIZES.lg,
    fontWeight: 'bold',
  },
  itemContainer: {
    paddingHorizontal: 10,
    borderRadius: 10,
    borderColor: COLORS.BLACK_OPAC_90,
    marginVertical: 8,
    borderWidth: 1,
  },
  item: {
    color: COLORS.BLACK,
    fontSize: FONT_SIZES.xl,
    marginVertical: 10,
  },
  seperator: {
    height: 0.5,
    width: WIDTH,
    backgroundColor: COLORS.BLACK_OPAC_90,
  },
});

export default CitySelector;
