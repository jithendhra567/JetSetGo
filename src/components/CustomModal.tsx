import React from 'react';
import {KeyboardAvoidingView, Modal, Pressable, ViewStyle} from 'react-native';
import {COLORS} from '../utils/styles';
import {HEIGHT, WIDTH} from '../utils/common';

type CustomModalProps = {
  visible: boolean;
  children?: React.ReactNode;
  toggleModal: (val: boolean) => void;
  dismissable?: boolean;
  presentationStyle?:
    | 'fullScreen'
    | 'pageSheet'
    | 'formSheet'
    | 'overFullScreen'
    | undefined;
  animationType?: 'slide' | 'fade' | 'none';
  backdropStyle?: ViewStyle;
};

export const CustomModal = ({
  visible,
  dismissable = true,
  toggleModal,
  children,
  presentationStyle,
  animationType = 'fade',
  backdropStyle,
}: CustomModalProps) => {
  return (
    <Modal
      animationType={animationType}
      transparent
      presentationStyle={presentationStyle}
      visible={visible}
      onDismiss={() => toggleModal(false)}
      onRequestClose={() => toggleModal(false)}>
      <Pressable
        style={[
          {
            width: WIDTH,
            height: HEIGHT,
            justifyContent: 'center',
            alignItems: 'center',
            backgroundColor: COLORS.BLACK_OPAC_50,
          },
          backdropStyle,
        ]}
        onPress={dismissable ? () => toggleModal(false) : undefined}>
        <KeyboardAvoidingView>{children}</KeyboardAvoidingView>
      </Pressable>
    </Modal>
  );
};

export default CustomModal;
