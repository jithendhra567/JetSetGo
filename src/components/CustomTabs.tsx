/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import {LayoutAnimation, StyleSheet, Text, View, ViewStyle} from 'react-native';
import {COLORS, FONT_SIZES} from '../utils/styles';

type Props = {
  tabs: string[];
  onTabPress: (index: number) => void;
  tabStyle?: ViewStyle;
  tabTextStyle?: ViewStyle;
  activeTabIndex: number;
};

export const CustomTabs = ({
  tabs,
  onTabPress,
  tabStyle,
  tabTextStyle,
  activeTabIndex,
}: Props) => {
  const onPress = (index: number) => {
    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
    onTabPress(index);
  };

  return (
    <View style={[styles.tabs, tabStyle]}>
      {tabs.map((tab, index) => (
        <View
          style={{
            alignItems: 'center',
            width: `${100 / tabs.length}%`,
            gap: 6,
          }}>
          <Text
            key={index}
            style={[
              styles.tabText,
              tabTextStyle,
              {opacity: index === activeTabIndex ? 1 : 0.5},
            ]}
            onPress={() => onPress(index)}>
            {tab}
          </Text>
          {activeTabIndex === index && (
            <View
              style={{
                height: 3,
                borderRadius: 10,
                width: '20%',
                backgroundColor: COLORS.ORANGE,
              }}
            />
          )}
        </View>
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  tabs: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  tabText: {
    color: COLORS.BLACK,
    fontSize: FONT_SIZES.xl,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
