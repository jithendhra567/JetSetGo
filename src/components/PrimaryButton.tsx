import React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';
import {COLORS} from '../utils/styles';

type Props = {
  text: string;
  onPress: () => void;
};

const PrimaryButton = ({text, onPress}: Props) => {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Text>{text}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    backgroundColor: COLORS.BLACK,
  },
});

export default PrimaryButton;
