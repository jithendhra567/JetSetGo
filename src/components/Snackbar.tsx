import React from 'react';
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withSpring,
} from 'react-native-reanimated';
import {StyleSheet, Text} from 'react-native';
import {COLORS, FONT_SIZES} from '../utils/styles';

export type SnackbarProps = {
  message?: string;
  duration?: number;
  type: 'info' | 'success' | 'error' | 'warning';
};

const configStyles = {
  info: {
    borderColor: COLORS.BLUE,
    borderWidth: 1,
    color: 'BLUE',
  },
  success: {
    borderColor: COLORS.GREEN,
    borderWidth: 1,
    color: 'GREEN',
  },
  error: {
    borderColor: COLORS.RED,
    borderWidth: 1,
    color: 'RED',
  },
  warning: {
    borderColor: COLORS.ORANGE,
    borderWidth: 1,
    color: 'ORANGE',
  },
};

const ANIMATION_DURATION = 900;

const DEFAULT_CONFIG: SnackbarProps = {
  message: '',
  duration: 3000,
  type: 'info',
};

const snackbarRef = React.createRef<any>();
const modalSnackRef = React.createRef<any>();

const Snackbar = React.forwardRef<any, {isModal?: boolean}>(
  ({isModal}, ref) => {
    const offset = useSharedValue(-100);
    const [data, setData] = React.useState<SnackbarProps>(DEFAULT_CONFIG);

    const showSnackbar = (snackData: SnackbarProps) => {
      if (!snackData.message) {
        return;
      }
      setData(snackData);
      offset.value = withSpring(100, {
        duration: ANIMATION_DURATION,
      });
      hideSnackbar(snackData.duration);
    };

    const hideSnackbar = (delay: number = 0) => {
      if (isModal) {
        snackbarRef.current?.hideSnackbar();
      }
      setTimeout(() => {
        offset.value = withSpring(-100, {
          duration: ANIMATION_DURATION,
        });
      }, delay);
    };

    React.useImperativeHandle(isModal ? ref : snackbarRef, () => ({
      showSnackbar,
      hideSnackbar,
    }));

    const animatedStyles = useAnimatedStyle(() => ({
      top: offset.value,
    }));

    return (
      <Animated.View
        ref={isModal ? ref : snackbarRef}
        style={[styles.container, configStyles[data.type], animatedStyles]}>
        <Text style={styles.text}>
          {typeof data.message === 'string'
            ? data.message
            : 'Something went wrong'}
        </Text>
        <Text style={styles.close} onPress={() => hideSnackbar(0)}>
          Close
        </Text>
      </Animated.View>
    );
  },
);

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    width: '90%',
    left: '5%',
    borderRadius: 6,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: 16,
    paddingVertical: 12,
    backgroundColor: COLORS.BLACK,
  },
  text: {
    width: '86%',
    color: COLORS.WHITE,
    fontSize: FONT_SIZES.lg,
  },
  close: {
    color: COLORS.WHITE,
    fontSize: FONT_SIZES.md,
  },
});

export const showSnackbar = (
  message: string,
  props: SnackbarProps = DEFAULT_CONFIG,
) => {
  const params = {
    ...DEFAULT_CONFIG,
    ...props,
    message,
  };
  snackbarRef.current?.showSnackbar(params);
  modalSnackRef.current?.showSnackbar(params);
};

export default Snackbar;
