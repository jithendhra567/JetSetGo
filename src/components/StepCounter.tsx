import React from 'react';
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {COLORS, FONT_SIZES} from '../utils/styles';

type Props = {
  steps: number;
  onStepsChange: (steps: number) => void;
};

const StepCounter = ({steps, onStepsChange}: Props) => {
  const onAdd = () => {
    onStepsChange(steps + 1);
  };

  const onSubtract = () => {
    if (steps === 0) {
      return;
    }
    onStepsChange(steps - 1);
  };

  return (
    <View style={styles.container}>
      <TouchableOpacity onPress={onSubtract} style={styles.actionContainer}>
        <Text
          style={[
            styles.action,
            {fontSize: FONT_SIZES['6xl'], transform: [{translateY: -4}]},
          ]}>
          -
        </Text>
      </TouchableOpacity>
      <Text style={styles.steps}>{steps}</Text>
      <TouchableOpacity onPress={onAdd} style={styles.actionContainer}>
        <Text style={[styles.action, {transform: [{translateY: -2}]}]}>+</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    gap: 20,
  },
  actionContainer: {
    borderRadius: 100,
    backgroundColor: COLORS.WHITE,
    borderWidth: 1,
    borderColor: COLORS.BORDER_COLOR,
    position: 'relative',
    padding: 16,
    justifyContent: 'center',
    alignItems: 'center',
  },
  action: {
    fontSize: FONT_SIZES['4xl'],
    position: 'absolute',
    color: COLORS.BLACK,
  },
  steps: {
    fontSize: FONT_SIZES['4xl'],
    color: COLORS.BLACK,
    fontWeight: 'bold',
  },
});

export default StepCounter;
