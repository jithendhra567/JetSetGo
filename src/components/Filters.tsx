import React, {useState} from 'react';
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {COLORS, FONT_SIZES} from '../utils/styles';
import {HEIGHT, WIDTH} from '../utils/common';
import PrimaryButton from './PrimaryButton';
import {CustomTabs} from './CustomTabs';

export type AirlineType = {
  name: string;
  length: boolean;
};

export type SortType = {
  price: number;
  duration: number;
  departure: number;
};

export type FiltersType = {
  sort: SortType;
  airlines: any;
};

type Props = {
  onApply: (data: FiltersType) => void;
  airlines: AirlineType[];
  onDismiss: () => void;
  filter: FiltersType;
};

const SORT: {
  key: 'price' | 'duration' | 'departure';
  value: number;
  text: string;
}[] = [
  {
    key: 'price',
    value: -1,
    text: 'Sort by Price (Low to High)',
  },
  {
    key: 'price',
    value: 1,
    text: 'Sort by Price (High to Low)',
  },
  {
    key: 'duration',
    value: -1,
    text: 'Sort by Duration (Shortest to Longest)',
  },
  {
    key: 'duration',
    value: 1,
    text: 'Sort by Duration (Longest to Shortest)',
  },
  {
    key: 'departure',
    value: -1,
    text: 'Sort by Departure Time (Earliest to Latest)',
  },
  {
    key: 'departure',
    value: 1,
    text: 'Sort by Departure Time (Latest to Earliest)',
  },
];

const Filters = ({filter, airlines = [], onApply, onDismiss}: Props) => {
  const [activeTabIndex, setActiveTabIndex] = useState(0);

  const [sort, setSort] = useState(filter.sort);

  const [selectedAirline, setSelectedAirline] = useState<any>(filter.airlines);

  const addSort = (key: keyof typeof sort, value: number) => {
    if (sort[key] === value) {
      setSort({
        ...sort,
        [key]: 0,
      });
      return;
    }
    setSort({
      ...sort,
      [key]: value,
    });
  };

  const onApplyFilter = () => {
    onApply({
      sort,
      airlines: selectedAirline,
    });
    onDismiss();
  };

  const renderContent = () => {
    if (activeTabIndex === 0) {
      return (
        <View style={{marginTop: 10}}>
          {SORT.map((item, index) => {
            const active =
              sort[item.key] !== 0 && sort[item.key] === item.value;
            return (
              <TouchableOpacity
                key={index}
                style={[
                  styles.sortItem,
                  {borderColor: active ? COLORS.BLUE : COLORS.GREY},
                ]}
                onPress={() => addSort(item.key, item.value)}>
                <Text
                  style={[
                    styles.sortText,
                    {color: active ? COLORS.BLUE : COLORS.BLACK},
                  ]}>
                  {item.text}
                </Text>
                <Text
                  style={[
                    styles.arrow,
                    {color: active ? COLORS.BLUE : COLORS.BLACK},
                  ]}>
                  {item.value < 0 ? '↓' : '↑'}
                </Text>
              </TouchableOpacity>
            );
          })}
        </View>
      );
    }
    return (
      <View style={{marginTop: 10}}>
        {airlines.map((item, index) => {
          const active = selectedAirline[item.name];
          return (
            <TouchableOpacity
              key={index}
              style={[
                styles.airLines,
                {borderColor: active ? COLORS.BLUE : COLORS.GREY},
              ]}
              onPress={() =>
                setSelectedAirline(
                  selectedAirline[item.name]
                    ? {
                        ...selectedAirline,
                        [item.name]: !selectedAirline[item.name],
                      }
                    : {...selectedAirline, [item.name]: true},
                )
              }>
              <Text
                style={[
                  styles.sortText,
                  {color: active ? COLORS.BLUE : COLORS.BLACK},
                ]}>
                {item.name}
              </Text>
              <Text
                style={{
                  fontSize: FONT_SIZES['2xl'],
                  fontWeight: 'bold',
                  color: active ? COLORS.BLUE : COLORS.BLACK,
                }}>
                {item.length}
              </Text>
            </TouchableOpacity>
          );
        })}
        {airlines.length === 0 && (
          <View
            style={{flex: 0.8, justifyContent: 'center', alignItems: 'center'}}>
            <Text style={{fontSize: FONT_SIZES['4xl'], fontWeight: 'bold'}}>
              No Airline Found
            </Text>
          </View>
        )}
      </View>
    );
  };

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.title}>Filters</Text>
        <TouchableOpacity onPress={onDismiss}>
          <Text style={styles.close}>Close</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.divider} />
      <View style={styles.filter}>
        <CustomTabs
          activeTabIndex={activeTabIndex}
          onTabPress={setActiveTabIndex}
          tabs={['Sort By', 'Filter']}
        />
        {renderContent()}
      </View>
      <PrimaryButton
        text="Apply"
        style={{
          width: WIDTH * 0.9,
          marginBottom: 20,
        }}
        onPress={onApplyFilter}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: HEIGHT * 0.9,
    width: WIDTH,
    alignItems: 'center',
    backgroundColor: COLORS.WHITE,
  },
  header: {
    width: '100%',
    paddingHorizontal: 20,
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
  },
  filter: {
    flex: 1,
  },
  title: {
    color: COLORS.BLACK,
    fontSize: FONT_SIZES['4xl'],
    fontWeight: 'bold',
    marginTop: 14,
    marginBottom: 14,
  },
  divider: {
    width: WIDTH,
    marginBottom: 14,
    height: 1,
    backgroundColor: COLORS.GREY,
  },
  close: {
    color: COLORS.RED,
    fontSize: FONT_SIZES.xl,
    fontWeight: 'bold',
  },
  sortItem: {
    paddingVertical: 12,
    borderWidth: 1,
    borderColor: COLORS.BORDER_COLOR,
    borderRadius: 10,
    marginHorizontal: 20,
    marginVertical: 10,
    paddingHorizontal: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  sortText: {
    color: COLORS.BLACK,
    fontSize: FONT_SIZES.lg,
    fontWeight: 'bold',
  },
  airLines: {
    paddingVertical: 10,
    borderWidth: 1,
    borderColor: COLORS.BORDER_COLOR,
    borderRadius: 10,
    marginHorizontal: 20,
    marginVertical: 10,
    paddingHorizontal: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  arrow: {
    color: COLORS.BLACK,
    fontSize: FONT_SIZES['5xl'],
    position: 'absolute',
    right: 14,
    top: -2,
    fontWeight: 'bold',
  },
});

export default Filters;
