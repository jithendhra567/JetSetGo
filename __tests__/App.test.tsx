/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import PrimaryButton from '../src/components/PrimaryButton';
import {CustomTabs} from '../src/components/CustomTabs';

test('primay button renders correctly', () => {
  const buttontree = renderer
    .create(<PrimaryButton text="test" onPress={() => {}} />)
    .toJSON();
  expect(buttontree).toMatchSnapshot();
});

test('custom tabs renders correctly', () => {
  const tabs = renderer
    .create(
      <CustomTabs
        tabs={['test', 'test2']}
        onTabPress={() => {}}
        activeTabIndex={0}
      />,
    )
    .toJSON();
  expect(tabs).toMatchSnapshot();
});
