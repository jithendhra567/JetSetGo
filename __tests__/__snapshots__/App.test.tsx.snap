// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom tabs renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "gap": 6,
        "opacity": 1,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#000",
            "fontSize": 53,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
          {
            "opacity": 1,
          },
        ]
      }
    >
      test
    </Text>
    <View
      style={
        {
          "backgroundColor": "#1E59F9",
          "borderRadius": 10,
          "height": 3,
          "width": "20%",
        }
      }
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "gap": 6,
        "opacity": 1,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#000",
            "fontSize": 53,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
          {
            "opacity": 0.5,
          },
        ]
      }
    >
      test2
    </Text>
  </View>
</View>
`;

exports[`primay button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#1E59F9",
      "borderRadius": 10,
      "elevation": 6,
      "opacity": 1,
      "paddingVertical": 14,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.19,
      "shadowRadius": 5.62,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#FFF",
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    test
  </Text>
</View>
`;
